Model/AdyenInitiateTerminalApi.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2018 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Model;
+
+use Adyen\Payment\Api\AdyenInitiateTerminalApiInterface;
+use Adyen\Payment\Model\Ui\AdyenPosCloudConfigProvider;
+use Adyen\Util\Util;
+
+class AdyenInitiateTerminalApi implements AdyenInitiateTerminalApiInterface
+{
+
+    private $_encryptor;
+    private $_adyenHelper;
+    private $_adyenLogger;
+    private $_recurringType;
+    private $_appState;
+
+    /**
+     * @var int
+     */
+    protected $storeId;
+
+    /**
+     * @var \Magento\Checkout\Model\Session
+     */
+    protected $_checkoutSession;
+
+    /**
+     * AdyenInitiateTerminalApi constructor.
+     * @param \Adyen\Payment\Helper\Data $adyenHelper
+     * @param \Adyen\Payment\Logger\AdyenLogger $adyenLogger
+     * @param \Magento\Checkout\Model\Session $_checkoutSession
+     * @param array $data
+     */
+    public function __construct(
+        \Adyen\Payment\Helper\Data $adyenHelper,
+        \Adyen\Payment\Logger\AdyenLogger $adyenLogger,
+        \Magento\Checkout\Model\Session $_checkoutSession,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        array $data = []
+    )
+    {
+        $this->_adyenHelper = $adyenHelper;
+        $this->_adyenLogger = $adyenLogger;
+        $this->_checkoutSession = $_checkoutSession;
+        $this->storeId = $storeManager->getStore()->getId();
+
+        // initialize client
+        $client = $this->_adyenHelper->initializeAdyenClient($this->storeId);
+        $apiKey = $this->_adyenHelper->getPosApiKey($this->storeId);
+        $client->setXApiKey($apiKey);
```

Comment:
Is the POS API key different from the API key? Maybe we could pass the api key as a parameter if we want to overwrite the default one like this:

```suggestion
        $apiKey = $this->_adyenHelper->getPosApiKey($this->storeId);
        $client = $this->_adyenHelper->initializeAdyenClient($this->storeId, $apiKey);
```
In Helper/Data.php
```
/**
	 * Initializes and returns Adyen Client and sets the required parameters of it
	 *
	 * @param int|null $storeId
         * @param string|null $apiKey
	 * @return \Adyen\Client
	 * @throws \Adyen\AdyenException
	 */
    public function initializeAdyenClient($storeId = null, $apiKey = null)
	{
                 if (!$apiKey) {
                      $apiKey = $this->getAPIKey($storeId);
                 }
		
                 // initialize client
		$client = $this->createAdyenClient();
		$client->setApplicationName("Magento 2 plugin");
		$client->setXApiKey($apiKey);

		$client->setAdyenPaymentSource($this->getModuleName(), $this->getModuleVersion());

		$client->setExternalPlatform($this->productMetadata->getName(), $this->productMetadata->getVersion());

		if ($this->isDemoMode($storeId)) {
			$client->setEnvironment(\Adyen\Environment::TEST);
		} else {
			$client->setEnvironment(\Adyen\Environment::LIVE, $this->getLiveEndpointPrefix($storeId));
		}

		$client->setLogger($this->adyenLogger);

		return $client;
	}
```